### ISSUE TITLE
Issue with stage 4 lightcurve modeling

### ISSUE THREAD

**Issue by suman-phy** [radicamc/exoTEDRF#21]
I'm facing a problem while running Stage 4 to fit the lightcurves using fit_lightcurves.py and fit_lightcurves.yaml. It is unable to fit the mid-transit time. There is always a large random shift in the time axis.

**Comment by radicamc** [comment]
Can you provide some more information about this? I have never encountered such an issue before.

**Comment by suman-phy** [comment]
I'm working on a NIRSpec/G395 transmission spectra. The stage1 to stage 3 of data reduction worked perfectly. However, at stage 4, I'm using the 'fit_lightcurves' files with the default set of model parameters, but it is never able to fit the lightcurves. There seems to be always some error with the mid-transit time (see the figure), and the shift is also not constant between each trial. 

![image](https://github.com/user-attachments/assets/3a9315c5-0fdb-4024-af51-ae385192226c)

**Comment by radicamc** [comment]
What do you mean by the "default set of model parameters"? Are you attempting to fit the mid-transit time, or fixing it?
Are you able to produce some minimal working example that reproduces this issue? That would be the easiest way to diagnose what is going wrong here.

**Comment by suman-phy** [comment]
I'm trying to fit the mid-transit time. Below is how my ''Fit Priors + Parameters'' look like.

`params : ['per_p1', 't0_p1', 'rp_p1', 'inc_p1',
          'u1_inst', 'u2_inst', 'ecc_p1', 'w_p1', 'a_p1',
          'sigma_inst']
dists : ['fixed', 'uniform', 'uniform', 'fixed',
         'uniform', 'uniform', 'fixed', 'fixed', 'fixed',
         'loguniform']
values : [3.3448285, [60010, 60012], [0.08, 0.16], 85.7,
          [0., 1.], [0., 1.], 0.0, 90., 7.17,
          [0.000001, 0.1]]`

The mid-transit time is approx. 60010.5. Below is the corner plot from the fit.

![image](https://github.com/user-attachments/assets/ace19181-7ce5-4e14-a93e-1237c133815c)

I've tried changing the priors widely and even ran on the other set, i.e. nrs2. However, the problem persists. It seems like the model isn't generating a transit function to fit the data, although I've selected `lc_model_type : 'transit'`

**Comment by radicamc** [comment]
No, I don't think the issue is that the code is producing an eclipse as opposed to a transit model. Have you tried fixing the mid-transit time to the value that you expect? What happens then?
Also, from where are you getting the expected mid-transit time? Does this come from a white light curve fit, or is it a literature value? If it's a literature value, keep in mind that the JWST timestamps are in MJD, so BJD-2450000.5. 
Also, where do the fixed a/R* and inclination values come from? Are they from the white light curve fit? How does the white light fit look?

**Comment by suman-phy** [comment]
Fixing the mid-transit doesn't change anything either (see below).

![image](https://github.com/user-attachments/assets/5766dfb0-4f39-42aa-b067-015bf3fb7fc8)

The expected mid-transit time is from the JWST time-stamps in MJD. The fixed a/R* and inclination values are from literature. I have also tried keeping them free, but that does not work either.

The white and chromatic lightcurve fits using Eureka! looks just fine (see below), which uses similar priors. However, I haven't been able to produce any correct fit using exoTEDRF yet.

![fig5101_ch00_lc_dynesty](https://github.com/user-attachments/assets/bbdecca1-329a-4fd2-b6bc-8b50abe71929)

**Comment by radicamc** [comment]
Interesting. 
Would you be willing to share with me the exact fit_lightcurves files that you're using as well as some example light curves (just a couple would be fine) so that I can try and reproduce this? It's difficult for me to diagnose what could be going wrong without knowing exactly what you're doing. If so, you can send everything to radicamc@uchicago.edu, or share whatever private google drive, dropbox etc, link that you wish.

**Comment by suman-phy** [comment]
Thanks Radica for extending your help. I'm in the midst of some deadlines in here now, so let me get back to you in a few days with the LCs and the model parameters. In the meantime, I have modeled the LCs using other packages, and I'm really impressed with the reduced data from exoTEDRF. Expect to see us using your pipeline in several works in future :)

**Comment by radicamc** [comment]
Closing this as there has been no updates in a while and I haven't been able to reproduce these issues.
